lib/galaxy/model/migrations/README.md

```diff
@@ -1,115 +1,391 @@
 
-Upgrade "galaxy" database to version "abc123" using default configuration:
+To accommodate this setup, Galaxy uses [Alembic
+branches](https://alembic.sqlalchemy.org/en/latest/branches.html#working-with-branches). A branch is
+a versioning lineage that starts at a common base revision and represents part of Galaxy's data
+model. These branches are identified by labels (***gxy*** for the galaxy model and ***tsi*** for the install
+model) and may share the same Alembic version table (if they share the same database; otherwise,
+each database has its own version table). Each branch has its own version history, represented by
+revision scripts located in the branch version directory (`migrations/alembic/versions_gxy` for
+*gxy* and `migrations/alembic/versions_tsi` for *tsi*). 
 
-`sh manage_db.sh upgrade --version=abc123`
+For a more detailed description of the system's internals, see pull request [#13108](https://github.com/galaxyproject/galaxy/pull/13108).
 
-Downgrade "install" database to version "xyz789" passing configuration file "mygalaxy.yml":
+## db.sh
 
-`sh manage_db.sh downgrade --version=xyz789 -c mygalaxy.yml install`
+This script offers a set of common database schema migration operations that are executed on the *gxy* branch, which, in the vast majority of cases, is all you need to develop and administer Galaxy.
+To run operations on the *tsi* branch, you need to use `run_alembic.sh`.
 
-Upgrade "galaxy" database to latest version using default configuration:
+```
+usage: db.sh [-h] [-c CONFIG] [--raiseerr] {upgrade,downgrade,version,v,
+       dbversion,dv,history,h,show,s,revision,init} ...
 
-`sh manage_db.sh upgrade`
+positional arguments:
+  {upgrade,downgrade,version,v,dbversion,dv,history,h,show,s,revision,init}
+    upgrade             Upgrade to a later version
+    downgrade           Revert to a previous version
+    version (v)         Show the head revision in the migrations script directory
+    dbversion (dv)      Show the current revision for Galaxy's database
+    history (h)         List revision scripts in chronological order
+    show (s)            Show the revision(s) denoted by the given symbol
+    revision            Create a new revision file
+    init                Initialize empty database(s) for both branches
+                        (create database objects for gxy and tsi branch)
 
-#### run_alembic.sh:
+optional arguments:
+  -h, --help            show this help message and exit
+  -c CONFIG, --galaxy-config CONFIG
+                        Alternate Galaxy configuration file
+  --raiseerr            Raise a full stack trace on error
+```
 
-##### Upgrading
+#### Revision identifiers
 
-Upgrade gxy to head revision:
+Some of the commands accept revision identifiers as arguments. A revision is usually identified by a
+12-digit hexadecimal number (i.e., `6a67bf27e6a6`). Anytime you need to refer to a specific
```

Comment:
I'd skip the partial number stuff, that's a nice thing but if you're reading the document I don't think that's relevant.

Reply:
Hmm.. I think it's helpful, especially for devs (given that we are now dealing with 12-digit hex numbers). But again, I'll move the description into a different section; the functionality will be there - it's how alembic processes revision identifiers.